src/err/mod.rs

```diff
@@ -982,6 +982,24 @@ impl PyErrArguments for PyDowncastErrorArguments {
     }
 }
 
+/// Python exceptions that can be converted to [`PyErr`].
+///
+/// This is used to implement [`From<Bound<'_, T>> for PyErr`].
+///
+/// Users should not need to implement this trait directly. It is implemented automatically in the
+/// [`crate::import_exception!`] and [`crate::create_exception!`] macros.
+pub trait ToPyErr {}
```

Comment:
👍 I will mull this name this afternoon and see how I feel about it come evening!

Reply:
I couldn't think of anything much better here, so let's go with that.

Reply:
As in the previous thread about `Deref`, I would suggest actually moving the operation into the trait and making the `From` impl a trivial wrapper to give the trait semantics instead of just being a marker.

For example, it could have a method like

```rust
fn to_py_err(self) -> PyErr;
```

and the macro could generate an impl for `Bound<$name>`. But it could also be implemented for e.g. `PyDowncastError` and other types which can be turned into an exceptions even if they do not start out as Python objects.

(I am somewhat surprised that we do not have such a trait yet, but I suspect we always use a plain `From` bound where we need to produce `PyErr` in generic context? I am not sure if we should lift `ToPyErr` to that position then or keep it an implementation detail.)

Reply:
I tried this, but you can see I'm getting errors related to orphan rules. Maybe I misunderstood your intended implementation?

Reply:
No, this is what I had in mind.

I understand where the orphan rule issues come from and I think fixing them would then imply that we cannot handle e.g. the `PyDowncastError` case using the resulting trait.

I guess we have to live with the pure marker trait for now. Sorry for leading you down this unproductive rabbit hole.